google/cloud/testing_util/environment_variable_restore.h

```diff
@@ -38,6 +38,8 @@ class EnvironmentVariableRestore {
     SetUp();
   }
 
+  ~EnvironmentVariableRestore() { TearDown(); }
```

Comment:
If the ctor/dtor are going to call `SetUp()`/`TearDown()` respectively, then perhaps `SetUp()` and `TearDown()` should not be public.

Reply:
I think we need them to be so these objects can be used as part of a test fixture?